src/main/node/node.ts

```diff
@@ -0,0 +1,101 @@
+import { exec } from 'child_process';
+import { type } from 'os';
+import { appendLog } from '../logger';
+
+class Platform {
+  public os: string;
+
+  constructor(os: string) {
+    this.os = os;
+  }
+
+  public static WINDOWS: Platform = new Platform('Windows_NT');
+
+  public static LINUX: Platform = new Platform('Linux');
+
+  public static MAC: Platform = new Platform('Darwin');
+
+  private static supportedPlatforms: Platform[] = [
+    Platform.WINDOWS,
+    Platform.LINUX,
+    Platform.MAC,
+  ];
+
+  public static Current(): Platform {
+    return new Platform(type());
+  }
+
+  public IsSupported(): boolean {
+    const index = Platform.supportedPlatforms.findIndex(
+      (platform) => platform.os === this.os
+    );
+    return index > -1;
+  }
+
+  public NodeManagerInstallationCommand(): string {
+    if (this === Platform.WINDOWS) return 'winget install nvs';
+    if (this === Platform.LINUX || this === Platform.MAC)
+      return 'npm install -g n';
+    return '';
+  }
```

Comment:
It would make sense semantically if this returned `string` or `undefined` when the platform does not match any supported ones.

Reply:
Done.